otj-server-core/src/main/java/com/opentable/server/SpringPortSelectionPostProcessor.java

```diff
@@ -0,0 +1,80 @@
+
+    static final String MANAGEMENT_SERVER_PORT = "management.server.port";
+    static final String JMX_PORT = "ot.jmx.port";
+    static final String HTTPSERVER_CONNECTOR_DEFAULT_HTTP_PORT = "ot.httpserver.connector.default-http.port";
+    static final String SERVER_PORT = "server.port";
+
+    @Override
+    public void postProcessEnvironment(final ConfigurableEnvironment environment, final SpringApplication application) {
+        Arrays.stream(environment.getActiveProfiles())
+                .filter("deployed"::equals)
+                .findFirst()
+                .map(i -> environment)
+                .ifPresent(this::injectPortSelectorPropertySource);
+    }
+
+    private void injectPortSelectorPropertySource(ConfigurableEnvironment environment) {
+        environment.getPropertySources()
+                .remove(PORT_SELECTOR_PROPERTY_SOURCE);
+        environment.getPropertySources()
+                .addLast(new PropertySource<Integer>(PORT_SELECTOR_PROPERTY_SOURCE) {
+                    @Override
+                    public Integer getProperty(@NonNull String s) {
+                        boolean isK8s = (!"IS_KUBERNETES".equalsIgnoreCase(s)) && "true".equalsIgnoreCase(environment.getProperty("IS_KUBERNETES", "false"));
+                        // Default spring boot
+                        if (SERVER_PORT.equalsIgnoreCase(s)) {
+                            return Integer.parseInt(environment.getProperty("PORT_HTTP",
+                                    environment.getProperty("PORT0", "0")));
+                        }
+                        // otj-server default connector
+                        if (HTTPSERVER_CONNECTOR_DEFAULT_HTTP_PORT.equalsIgnoreCase(s)) {
+                            return Integer.parseInt(environment.getProperty("PORT_HTTP",
+                                    environment.getProperty("PORT0", "-1")));
+                        }
+                        // otj-server named connector
+                        if (isK8s && s.matches("ot\\.httpserver\\.connector\\..*-.*\\.port")) {
+                            final String name = s.split("\\.")[3].split("-")[0].toUpperCase(Locale.US);
+                            return Integer.parseInt(environment.getProperty("PORT_" + name, "-1"));
+                        }
+                        // jmx
+                        if (JMX_PORT.equalsIgnoreCase(s)) {
+                            return Integer.parseInt(environment.getProperty("PORT_JMX",
+                                    environment.getProperty("PORT1", "0")));
+                        }
+                        // actuator
+                        if (MANAGEMENT_SERVER_PORT.equalsIgnoreCase(s)) {
+                            return Integer.parseInt(environment.getProperty("PORT_ACTUATOR",
+                                    environment.getProperty("PORT2", "0")));
+                        }
+                        return null;
+                    }
+                });
```

Comment:
Log the built property source values. In fact frankly I don't see why you don't statically set this. This seems very weird